sqlglot/dialects/tsql.py

```diff
@@ -339,6 +340,16 @@ def _json_extract_sql(
     return self.func("ISNULL", json_query, json_value)
 
 
+def _timestrtotime_sql(self: TSQL.Generator, expression: exp.TimeStrToTime):
+    sql = timestrtotime_sql(self, expression)
+    if expression.args.get("timezone"):
+        # If there is a timezone, produce an expression like:
+        # CAST('2020-01-01 12:13:14-08:00' AS DATETIMEOFFSET) AT TIME ZONE 'UTC'
+        # If you dont have AT TIME ZONE 'UTC', wrapping that expression in another cast back to DATETIME2 just drops the timezone information
```

Comment:
Why does it drop the timezone information? What does the `AT TIME ZONE 'UTC'` achieve here?

Reply:
Because it's Microsoft so therefore insane. Based on my testing:

```
select CAST('2020-01-01 12:13:14-08:00' AS DATETIMEOFFSET)
> 2020-01-01 12:13:14.000 -0800
```
ok, fine, its showing the timezone. What if I want it as `DATETIME2` instead?

```
select CAST(CAST('2020-01-01 12:13:14-08:00' AS DATETIMEOFFSET) AS DATETIME2)
> 2020-01-01 12:13:14.000
```
_Of course_ it just strips the timezone off, totally logical

```
select CAST(CAST('2020-01-01 12:13:14-08:00' AS DATETIMEOFFSET) AT TIME ZONE 'UTC' AS DATETIME2)
> 2020-01-01 20:13:14.000
```
Magic sauce to get the expected value

Reply:
Fun :-)

FWIW I did find another [thread](https://learn.microsoft.com/en-us/answers/questions/659376/converting-datetimeoffset-to-datetime-with-adjustm) where the `AT TIME ZONE 'UTC'` trick is mentioned. The problem stated by the OP looks similar to this one, so... Seems legit?

Reply:
Heh, that thread is even worse. They had to convert to UTC first before converting to the actual timezone they wanted.

I guess this makes the conversion to UTC automatic by default so things downstream should Just Work™